torch/csrc/jit/codegen/cuda/lower_allocation.cpp

```diff
@@ -519,38 +513,64 @@ class AllocationInserter : public kir::KirVisitor {
 
       AllocationInformation allocation;
       allocation.buffer = out_tv;
-      findAllocationPosition(allocation, expr);
-      createAllocExpr(allocation, is_output);
-      createInitExpr(allocation, init);
+      fillAllocationInformation(allocation, expr);
+
+      auto alloc_expr = createAllocExpr(allocation, is_output);
+      auto init_expr = createInitExpr(allocation, init);
 
       // Write information to GPULower
-      writeInfoToGPULower(allocation);
+      writeInfoToGPULower(allocation, alloc_expr);
+
+      if (alloc_expr != nullptr) {
+        if (allocation.buffer->memoryType() == MemoryType::Shared) {
+          // Shared allocations go at the begining of scope
+          TORCH_INTERNAL_ASSERT(!exprs_.empty());
+          registerInsertBefore(exprs_[0], alloc_expr, nullptr);
+        } else {
+          TORCH_INTERNAL_ASSERT(allocation.alloc_place_before != nullptr);
+          // Want to add exprs in reverse order, so we insert before the
```

Comment:
nit: the comment was truncated?

Reply:
Should just be removed. We used to insert first the alloc, then the insertion, but with insert before, we can do this naturally with forward placed insertions. Will remove.